torch/distributed/pipelining/microbatch.py

```diff
@@ -115,6 +116,111 @@ class _Replicate:
     pass
 
 
+def _split_block_mask(
+    block_mask: BlockMask,
+    spec: TensorChunkSpec,
+    num_chunks: int,
+) -> list[BlockMask]:
+    """Given a block mask, and a chunking spec, split the block mask.
+
+    Args:
+        block_mask: Block mask to split
+        spec: Chunking spec
```

Comment:
is there a case where we would want to specify on non-0 dimension for block mask? If it's always 0, then maybe passing the spec is not necessary

Reply:
sound good